autogpt_platform/backend/backend/blocks/slack/_api.py

```
@@ -7,13 +7,15 @@
 
 import logging
-from typing import Any, Optional
+from typing import Any, Optional, Union
 
 from pydantic import BaseModel
 
-from backend.data.model import APIKeyCredentials
+from backend.data.model import APIKeyCredentials, OAuth2Credentials
 from backend.util.request import Requests
 
+SlackCredentials = Union[APIKeyCredentials, OAuth2Credentials]
```

Ruff (0.16.0)

[warning] Use `X | Y` for type annotations

(UP007)